src/controllers/project.ts

```diff
@@ -90,136 +46,87 @@ export const getProject = catchAsync(async (req, res, next) => {
 
   const project = await DB.Project.findOne({ projectName });
 
+  if (!project) {
+    return next(createError(400, "Project data does not exist."));
+  }
+
   return res.json({
     message: "ok",
     result: project,
   });
 });
 
 export const updateProject = catchAsync(async (req, res, next) => {
-  const { preject_name: projectName } = req.params;
-  const { type } = req.query;
-
-  switch (type) {
-    case "pull_request_update": {
-      const { action, pull_request, repository } = req.body;
-
-      if (action !== "closed" || !pull_request.merged) {
-        return next(
-          createError(
-            401,
-            "Received abnormal PR data or PR that has not yet been merged.",
-          ),
-        );
+  const { project_name: projectName } = req.params;
+  const { event, payload } = res.locals;
+
+  switch (event) {
+    case "ping": {
+      return res.json({
+        message: "The webhook has been successfully installed.",
+      });
+    }
+    case "push": {
+      if (
+        payload.ref.slice(11) !== "main" &&
+        payload.ref.slice(11) !== "master"
+      ) {
+        return res.status(304).json({
+          message: "ok",
+        });
       }
 
-      const githubAccessToken = Config.USER_CREDENTIAL_TOKEN;
-      const githubClient = new GithubClient(githubAccessToken as string);
+      const { head_commit: headCommit, repository } = req.body;
+
+      const webhookId = repository.id;
+      const projectUpdatedAt = headCommit.timestamp;
+      const lastCommitMessage = headCommit.message;
+      const lastCommitHash = headCommit.id;
+
       const project = new ProjectService();
 
-      const repoOwner = repository.owner.login;
-      const repoName = repository.name;
-      const headRef = pull_request.head.ref;
+      await project.updateProject({
+        webhookId,
+        projectUpdatedAt,
+        lastCommitMessage,
+        lastCommitHash,
+      });
```

Comment:
```suggestion
      await project.updateProject({
        webhookId: repository.id,
        projectUpdatedAt: headCommit.timestamp,
        lastCommitMessage: headCommit.message,
        lastCommitHash:  headCommit.id,
      });
```

개인적인 의견인데, 저 값들은 따로 변수화해주지 않아도 될 것 같아요..! 객체 구조분해할당한 데이터가 바로 단축 프로퍼티로 들어가는 경우가 아니고 단축 프로퍼티를 만들기위해 별도의 변수선언을 하는 경우라서요.

바꾼 코드 suggestion으로 남겼는데 좀 더 선호하시는 방향으로 하시면 될 것 같아요!

Reply:
그렇네요 좀 더 코드 수를 줄이면서도 명시적으로 보이게 할 수 있겠네요! 변경하겠습니다.